src/libosd/include/osd/cl_dem_uart.h

```diff
@@ -0,0 +1,116 @@
+/* Copyright 2018 The Open SoC Debug Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef OSD_CL_DEM_UART_H
+#define OSD_CL_DEM_UART_H
+
+#include <stdlib.h>
+
+#include <osd/osd.h>
+#include <osd/packet.h>
+#include <osd/hostmod.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+
+/**
+ * A osd_dem_uart_event represents one or more characters.
+ */
+struct osd_dem_uart_event {
+	uint8_t num_chars;
+	uint8_t chars[0];
+};
+
+typedef void (*osd_cl_dem_uart_handler_fn)(
+	void * /*cb_arg*/, struct osd_dem_uart_event * /*event to be handled*/);
+
+struct osd_dem_uart_event_handler {
+	osd_cl_dem_uart_handler_fn cb_fn;
+	void *cb_arg;
+};
+
+struct osd_dem_uart_desc {
+	unsigned int di_addr;
```

Comment:
I know we are not consistent across whole OSD, but please be consistent in your class at least: either use `unsigned int` for a DI address, or use `uint16_t` (c.f. `osd_cl_dem_uart_get_desc`)